crates/proof/src/cert_validity.rs

```diff
@@ -0,0 +1,44 @@
+use alloy_primitives::B256;
+use eigenda_v2_struct_rust::EigenDAV2Cert;
+use risc0_zkvm::is_dev_mode;
+use risc0_zkvm::Receipt;
+
+#[derive(Debug, Clone, Default)]
+pub struct CertValidity {
+    /// the claim about if the cert is valid
+    pub claimed_validity: bool,
+    /// a zkvm proof attesting the above result
+    /// in dev mode, receipt is ignored
+    pub receipt: Option<Receipt>,
+}
+
+impl CertValidity {
+    /// verify if the receipt for cert is valid
+    /// note this is different from if the cert itself is valid as in the is_valid field
+    pub fn validate_cert_receipt(
+        &self,
+        eigenda_cert: &EigenDAV2Cert,
+        validity_call_verifier_id: B256,
+    ) {
```

Comment:
if you define this function on `Receipt` directly and make it return `is_valid`, then I think we can get rid of the CertValidity struct altogether.